packages/core/content-releases/server/src/controllers/release.ts

```diff
@@ -29,50 +33,106 @@ const releaseController = {
-    if (isFindManyForContentTypeEntry) {
-      const query: GetContentTypeEntryReleases.Request['query'] =
-        await permissionsManager.sanitizeQuery(ctx.query);
+    const query = await permissionsManager.sanitizeQuery(ctx.query);
+    const { contentTypeUid, documentId, hasEntryAttached, locale } = query;
+    const entryId = await getEntryId({ contentTypeUid, documentId, locale }, { strapi });
 
-      const contentTypeUid = query.contentTypeUid;
-      const entryId = query.entryId;
-      // Parse the string value or fallback to a default
-      const hasEntryAttached: GetContentTypeEntryReleases.Request['query']['hasEntryAttached'] =
-        typeof query.hasEntryAttached === 'string' ? JSON.parse(query.hasEntryAttached) : false;
+    const isEntryAttached =
+      typeof hasEntryAttached === 'string' ? Boolean(JSON.parse(hasEntryAttached)) : false;
 
-      const data = hasEntryAttached
-        ? await releaseService.findManyWithContentTypeEntryAttached(contentTypeUid, entryId)
-        : await releaseService.findManyWithoutContentTypeEntryAttached(contentTypeUid, entryId);
+    if (isEntryAttached) {
+      const releases = await releaseService.findMany({
+        where: {
+          actions: {
+            target_type: contentTypeUid,
+            target_id: entryId,
+          },
+          releasedAt: {
+            $null: true,
+          },
+        },
+        populate: {
+          actions: {
+            fields: ['type'],
+            where: {
+              target_type: contentTypeUid,
+              target_id: entryId,
+            },
+          },
+        },
+      });
 
-      ctx.body = { data };
+      ctx.body = { data: releases };
     } else {
-      const query: GetReleases.Request['query'] = await permissionsManager.sanitizeQuery(ctx.query);
-      const { results, pagination } = await releaseService.findPage(query);
-
-      const data = results.map((release: ReleaseWithPopulatedActions) => {
-        const { actions, ...releaseData } = release;
-
-        return {
-          ...releaseData,
+      const relatedReleases = await releaseService.findMany({
+        where: {
+          releasedAt: {
+            $null: true,
+          },
           actions: {
-            meta: {
-              count: actions.count,
-            },
+            target_type: contentTypeUid,
+            target_id: entryId,
           },
-        };
+        },
       });
 
-      const pendingReleasesCount = await strapi.db.query(RELEASE_MODEL_UID).count({
+      const releases = await releaseService.findMany({
         where: {
-          releasedAt: null,
+          $or: [
+            {
+              id: {
+                $notIn: relatedReleases.map((release: any) => release.id),
+              },
+            },
+            {
+              actions: null,
+            },
+          ],
+          releasedAt: {
+            $null: true,
+          },
         },
       });
```

Comment:
This is because of the logic expected by the query engine.

Let's say you want to get the releases where documentId `abcd` is not attached 

If I say something like:
```
strapi.db("releases").query({
  where: {
    actions: {
      entryDocumentId: not abcd,
    }
  }
})
```

This could return releases where the documentId `abcd` is attached, because if you have two documents in a release, one is `abcd` and the other is anything, that other makes the condition pass